ruddock/modules/budget/helpers.py

```diff
@@ -0,0 +1,296 @@
+import flask
+import sqlalchemy
+import enum
+
+from datetime import datetime #ugh
+from decimal import Decimal
+
+# Enum for payment types
+class PaymentType(enum.Enum):
+  CASH = "Cash"
+  CHECK = "Check"
+  DEBIT = "Debit"
+  ONLINE = "Online"
+  TRANSFER = "Transfer"
+  OTHER = "Other"
+
+def get_today():
+  return datetime.now().strftime("%Y-%m-%d")
+
+def stringify(db_rows):
+  """Changes all columns to strings.
+     This is especially useful when working with NUMERIC, because
+     JSON refuses to serialize Decimal objects.
+     This has the side effect of replacing the RowProxy objects with
+     dictionaries, but hopefully no method should notice... :/ """
+
+  # maybe have it detect numeric columns?
+
+  rows = [dict(r.items()) for r in db_rows]
+
+  for r in rows:
+    for k in r:
+      r[k] = str(r[k])
+
+  return rows
+
+def get_payment_types():
+  return [x.value for x in PaymentType]
+
+def is_delayed_type(val):
+  """Returns true if this is a kind of payment that doesn't take effect immediately.
+     In other words, is this a check?"""
+  return val == PaymentType.CHECK.value
+
+def get_fyears():
+  """Gets list of all fiscal years."""
+  query = sqlalchemy.text("""
+    SELECT fyear_id, fyear_num
+    FROM budget_fyears
+    ORDER BY fyear_id DESC
+    """)
+
+  return flask.g.db.execute(query).fetchall()
+
+def get_fyear_from_num(fyear_num):
+  """Looks up the record for the given year.
+     If fyear_num is None, then this returns the current year.
+     If no matching row is found, returns None."""
+
+  if fyear_num is None:
```

Comment:
Would be more readable to just have two separate functions:

get_fyear(fyear_num)
get_current_fyear()